service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/SempDeleteCommandManager.java

```diff
@@ -0,0 +1,271 @@
+                    .status(JobStatus.success)
+                    .logs(List.of())
+                    .build());
+        } catch(ApiException e) {
+            ApiException apiException = (ApiException) e;
+            log.error("Error executing SEMP delete command: {}", apiException.getResponseBody());
+            setCommandError(command, e);
+        } catch (Exception e) {
+            log.error(ERROR_EXECUTING_COMMAND, e);
+            setCommandError(command, e);
+        }
+    }
+
+
+    private void executeSempDeleteCommand(Command command, SempApiProvider sempApiProvider) throws ApiException, JsonProcessingException {
+        SempEntityType deletionEntityType = SempEntityType.valueOf((String) command.getParameters().get(SempDeleteCommandConstants.SEMP_DELETE_ENTITY_TYPE));
+        switch (deletionEntityType) {
+            case solaceAclProfile:
+                executeDeleteAclProfile(command, sempApiProvider);
+                break;
+            case solaceQueue:
+                executeDeleteSolaceQueue(command, sempApiProvider);
+                break;
+            case solaceQueueSubscriptionTopic:
+                executeDeleteSolaceQueueTopicSubscription(command, sempApiProvider);
+                break;
+            case solaceClientUsername:
+                executeDeleteClientUsername(command, sempApiProvider);
+                break;
+            case solaceAuthorizationGroup:
+                executeDeleteAuthorizationGroup(command, sempApiProvider);
+                break;
+            case solaceAclSubscribeTopicException:
+                executeDeleteAclSubscribeTopicException(command, sempApiProvider);
+                break;
+            case solaceAclPublishTopicException:
+                executeDeleteAclPublishTopicException(command, sempApiProvider);
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported entity type for deletion: " + deletionEntityType);
+        }
+    }
+
+    private void executeDeleteAclProfile(Command command, SempApiProvider sempApiProvider) throws ApiException, JsonProcessingException {
+        AclProfileApi aclProfileApi = sempApiProvider.getAclProfileApi();
+        SempAclProfileDeletionRequest request = objectMapper.readValue(
+                objectMapper.writeValueAsString(command.getParameters().get(SEMP_DELETE_DATA)),
+                SempAclProfileDeletionRequest.class);
+        Validate.notEmpty(request.getMsgVpn(), "Msg VPN must not be empty");
+        Validate.notEmpty(request.getAclProfileName(), "ACL profile name must not be empty");
+
+        log.info("Deleting ACL profile via Semp V2 delete: {}", request.getAclProfileName());
+        try {
+            aclProfileApi.deleteMsgVpnAclProfile(request.getMsgVpn(), request.getAclProfileName());
+        } catch (ApiException e) {
+            // If the client username does not exist, we don't want to consider it an error
```

Comment:
I think the method `executeDeleteClientUsername` was copy pasted few times since this exact comment shows up 5 times in this PR. 

That's a good indication of being able to write a helper method that all these 5 `execteDelete{something}` methods can call.

You can ask github co-pilot to extract the common code from these 5 methods and refactor them for you.